test/databaseTest.js

```diff
@@ -108,7 +104,47 @@ describe('database', function(){
 
 
     function countConnections(db) {
-      return db.executeSql('SELECT count(*) as connections FROM pg_stat_activity')
+      return db.executeSql('SELECT count(*) as connections FROM pg_stat_activity WHERE application_name=$1',
+        [boss.db.config.application_name])
+        .then(result => parseFloat(result.rows[0].connections));
+    }
+
+  });
+
+  it('connection count does not exceed configured pool size with `max`', function(finished){
```

Comment:
I'd prefer to move this into a config test (does poolSize == max) instead of duplicating the entire connection pooling test

Reply:
added new tests, do you like it that way? should we remove the pooling test?

Reply:
Thanks!  I'll go ahead and merge and take a stab at the migration failure rolllback test